CMakeLists.txt

```diff
@@ -114,6 +117,41 @@ if(CODECOVERAGE)
     message( "-- Building with llvm Code Coverage Tools")
 endif()
 
+if(CPPCHECK)
+    find_program(cppcheck cppcheck HINTS "/usr/local/bin/cppcheck" REQUIRED)
+    message(STATUS "cppcheck ${cppcheck}")
+    if(cppcheck MATCHES "NOTFOUND")
+      message(FATAL_ERROR "failed to find cppcheck executable for CPPCHECK option")
+    endif()
+    # Create <cppcheck> work folder for whole program analysis, for faster analysis and to store some useful debug information
+    # Add cppcheck work folder and reports folder for cppcheck output.
+    file(MAKE_DIRECTORY ${PROJECT_BINARY_DIR}/cppcheck/reports/)
+    # max number of threads = number of CPUs
+    include(ProcessorCount)
+    ProcessorCount(CPU_CORES)
+    # Report output is sent to stderr. To have the output in a single report file, run run-cppcheck.sh only to analyse results, for example:
+    # concord-bft/build > make bcstatetransfer_tests 2> cppcheck/reports/full_report.txt
+    # concord-bft/build > scripts/run-cppcheck.sh -p
+    set(CMAKE_CXX_CPPCHECK
+          "${cppcheck}"
+          "--enable=all"
+          "--inconclusive"
+          "--inline-suppr"
+          "--quiet"
+          "--std=c++17"
+          "--platform=unix64"
```

Comment:
Since we aim to a multiplatform project, I wouldn't mention a platform here

Reply:
removed here..keeping in script